lib/import/colorectal/providers/st_george/st_george_handler_colorectal.rb

```diff
@@ -0,0 +1,303 @@
+        class StGeorgeHandlerColorectal < Import::Germline::ProviderHandler
+          include Import::Helpers::Colorectal::Providers::Rj7::Constants
+
+          def process_fields(record)
+            genotype = Import::Colorectal::Core::Genocolorectal.new(record)
+
+            # records using new importer should only have SRIs starting with V
+            return unless record.raw_fields['servicereportidentifier'].start_with?('V')
+
+            # add standard passthrough fields to genotype object
+            genotype.add_passthrough_fields(record.mapped_fields, record.raw_fields,
+                                            PASS_THROUGH_FIELDS)
+
+            assign_test_scope(genotype, record)
+            genotypes = fill_genotypes(genotype, record)
+
+            genotypes.each do |single_genotype|
+              process_variants(single_genotype, record)
+
+              @persister.integrate_and_store(single_genotype)
+            end
+          end
+
+          def assign_test_type(genotype, record)
+            # extract molecular testing type from the raw record
+            # map molecular testing type and assign to genotype using
+            # add_molecular_testing_type_strict method from genocolorectal.rb
+
+            return if record.raw_fields['moleculartestingtype'].blank?
+
+            return unless TEST_TYPE_MAP[record.raw_fields['moleculartestingtype']&.downcase&.strip]
+
+            genotype.add_molecular_testing_type_strict(TEST_TYPE_MAP[record.raw_fields['moleculartestingtype']&.downcase&.strip])
+          end
+
+          def assign_test_scope(genotype, record)
+            # extract molecular testing type from the raw record
+            # map molecular testing type and assign to genotype using
+            # add_test_scope method from genocolorectal.rb
+            # return (filled) genotype
+
+            testscope = record.raw_fields['moleculartestingtype']&.downcase&.strip
+            genotype.add_test_scope(TEST_SCOPE_MAP[testscope])
+            return genotype if genotype.attribute_map['genetictestscope'].present?
+
+            @logger.error 'ERROR - record with no genetic test scope, ask Fiona for new rules'
+          end
+
+          def fill_genotypes(genotype, record)
+            # process the genes, genotypes and test status for each gene listed in the genotype
+
+            genes_dict = process_genes(record)
+            handle_test_status(record, genotype, genes_dict)
+          end
+
+          def process_genes(record)
+            # process the gene names in columns listed and check for matches with the CRC gene regex list
+            # add matched tgenes to the genes_dict
+            # if no matched to crc gene regex list then check for matches in CRC gene map to add to the genes_dict
+            # return genes_dict
+            genes_dict = {}
+
+            ['gene', 'gene (other)', 'variant dna', 'test/panel'].each do |column|
```

Comment:
@NImeson , Your are looking for genes in all columns irrespective of test scope, but for targeted rules say to pick only from `Get gene from input column raw:gene or raw:gene (other)`

Reply:
edited method in [d16287a](https://github.com/NHSDigital/data_management_system/pull/81/commits/d16287ae4da6bcd9dea21fce634bc34266d362ddl)